ext/opcache/jit/README.md

```diff
@@ -24,5 +24,116 @@ was almost 100 times slower, making it prohibitively expensive to use.
+```
+
+- `opcache.jit_buffer_size=16M` enables the JIT in tests by providing 16 megabytes of
+  memory to use with the JIT to test with.
+- `opcache.protect_memory=1` will detect writing to memory that is meant to be
+  read-only, which is sometimes the cause of opcache bugs.
+- `--repeat 2` is optional, but used in CI since some JIT bugs only show up after processing a
+  request multiple times, e.g. due to a bug in the engine or bad data in the
+  runtime cache of the interpreter.
+- `-j$(nproc)` runs as many workers to run tests as there are CPUs.
+- `ext/opcache/` is the folder with the tests to run, in this case opcache
+  itself.  If no folders are provided, all tests are run.
+
+When investigating test failures such as segmentation faults, adding
+`-m --show-mem` may be useful to test with [valgrind](https://valgrind.org/) to detect out of bounds memory accesses.
+
+Note that the JIT supports 3 different architectures: `X86_64`, `i386`, and `arm64`.
+
+Miscellaneous
+-------------
+
+### Checking dasc files for in a different architecture
+
+The following command can be run to manually check if the modified `.dasc code` is at least transpilable
+for an architecture you're not using, e.g.:
+
+For arm64: `ext/opcache/minilua ext/opcache/jit/dynasm/dynasm.lua -D ARM64=1 -o ext/opcache/jit/zend_jit_arm64.ignored.c ext/opcache/jit/zend_jit_arm64.dasc`
+
+For x86_64: `ext/opcache/minilua ext/opcache/jit/dynasm/dynasm.lua -D X64=1 -o ext/opcache/jit/zend_jit_x86.ignored.c ext/opcache/jit/zend_jit_x86.dasc`
+
+For i386 (i.e. 32-bit): `ext/opcache/minilua ext/opcache/jit/dynasm/dynasm.lua  -o ext/opcache/jit/zend_jit_x86.ignored.c ext/opcache/jit/zend_jit_x86.dasc`
+
+### How to build 32-bit builds on x86_64 environments
+
+Refer to [../../../azure/i386](../../../azure/i386/apt.yml) for examples of
+dependencies to install.
+
+If you are running this natively (outside of Docker or a VM):
+
+- Consider running in docker/a VM instead if you are unfamiliar with this.
+- Avoid purging packages.
+- Avoid `-y` - if the package manager warns you that the dependencies conflict
+  then **don't** try to force install them.
+
+#### Prerequisites for 32-bit builds
+
+This assumes you are using a Debian-based Linux distribution and have already
+set up prerequisites for regular development.
+
+```
+sudo dpkg --add-architecture i386
+sudo apt-get update -y
+# As well as anything else from azure/i386/apt.yml that you're testing locally
+sudo apt-get install \
+    gcc-multilib g++-multilib \
+    libxml2-dev:i386 \
+    libc6:i386
+```
+
+#### Compiling 32-bit builds
+
+This assumes you are using a Debian-based Linux distribution and have already
+set up prerequisites for 32-bit development.
+
+```
+export LDFLAGS=-L/usr/lib/i386-linux-gnu
+export CFLAGS='-m32'
+export CXXFLAGS='-m32'
```

Comment:
This line probably isn't needed.

Reply:
It's copied from azure/i386/job.yml - it'll prevent issues if anyone adapts these instructions to `--enable-intl`. ./ext/intl/breakiterator/breakiterator_class.cpp

Reply:
I believe our CFLAGS also applies to C++ code, but I can't be bothered to double check...

Reply:
```make
ext/intl/intl_convertcpp.lo: /path/to/php-src/ext/intl/intl_convertcpp.cpp
	$(LIBTOOL) --mode=compile $(CXX) -Iext/intl/ -I/path/to/php-src/ext/intl/ $(COMMON_FLAGS) $(CXXFLAGS_CLEAN) $(EXTRA_CXXFLAGS)  -DU_NO_DEFAULT_INCLUDE_UTF_HEADERS=1 -DU_HIDE_OBSOLETE_UTF_OLD_H=1 -Wno-write-strings -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -std=c++11  -DUNISTR_FROM_CHAR_EXPLICIT=explicit -DUNISTR_FROM_STRING_EXPLICIT=explicit -c /path/to/php-src/ext/intl/intl_convertcpp.cpp -o ext/intl/intl_convertcpp.lo  -MMD -MF ext/intl/intl_convertcpp.dep -MT ext/intl/intl_convertcpp.lo

main/strlcpy.lo: /path/to/php-src/main/strlcpy.c
	$(LIBTOOL) --mode=compile $(CC) -Imain/ -I/path/to/php-src/main/ $(COMMON_FLAGS) $(CFLAGS_CLEAN) $(EXTRA_CFLAGS) -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -c /path/to/php-src/main/strlcpy.c -o main/strlcpy.lo  -MMD -MF main/strlcpy.dep -MT main/strlcpy.lo
```

If I add `CFLAGS=-Wunused` and then run `./config.nice`, the only place `-Wunused` gets added is CFLAGS_CLEAN. 


```make
CFLAGS = $(CFLAGS_CLEAN) -prefer-non-pic -static
CFLAGS_CLEAN = -fno-common -Wformat-truncation -Wlogical-op -Wduplicated-cond -Wno-clobbered -Wall -Wextra -Wno-strict-aliasing -Wno-unused-parameter -Wno-sign-compare -Wunused -fvisibility=hidden -O0 -Wimplicit-fallthrough=1 -g -DZEND_SIGNALS $(PROF_FLAGS)
CPP = cc -E
CPPFLAGS = -D_GNU_SOURCE
CXX = g++
CXXFLAGS = -g -O0 -prefer-non-pic -static $(PROF_FLAGS)
CXXFLAGS_CLEAN = -g -O0
```